src/bin/pg_autoctl/cli_do_fsm.c

```diff
@@ -274,15 +274,20 @@ keeper_cli_fsm_assign(int argc, char **argv)
 			break;
 		}
 
-		case 3:
+		case 4:
 		{
+			/* we just accept a single standby at a time here */
+			NodeAddress *otherNode = &(keeper.otherNodes.nodes[0]);
+
+			keeper.otherNodes.count = 1;
+
 			goalState = NodeStateFromString(argv[0]);
 
-			/* now prepare host and port in keeper.otherNode */
-			strlcpy(keeper.otherNode.host, argv[1], _POSIX_HOST_NAME_MAX);
+			/* now prepare host and port in keeper.otherNodes */
```

Comment:
id parameter is expected to be before host and port. We are still getting those from their original offsets

Reply:
and doing nothing regarding the id parameter

Reply:
Good spot, fixed.